legal-api/src/legal_api/resources/v2/business/colin_sync.py

```diff
@@ -287,7 +287,8 @@ def _set_shares(primary_or_holding_business, amalgamation_filing, transaction_id
     share_classes = VersionedBusinessDetailsService.get_share_class_revision(transaction_id,
                                                                              primary_or_holding_business.id)
     amalgamation_filing['shareStructure'] = {'shareClasses': share_classes}
-    business_dates = [item.resolution_date.isoformat() for item in primary_or_holding_business.resolutions]
+    resolutions = Resolution.get_resolution_by_business_id(primary_or_holding_business.id)
```

Comment:
This is a temporary fix to handle the missing relationship in versioned objects. 
I've tested both scenarios (FF ON/OFF), and everything works fine.

But I'm not sure if this approach is acceptable. 
I only found this issue in this ticket and am unsure if it exists elsewhere. (Didn't find this issue from the previous submissions of tickets)

Should we:
1. Investigate and fix our db versioning to handle relationships properly
2. Consider how this change might impact our already completed tickets that use versioning

@leodube-aot would appreciate your thoughts on this!

Reply:
Seems like this is an issue we have been aware of, but I don't think it is too prevalent in the code.

<img width="844" alt="Screenshot 2024-12-05 at 9 52 09 AM" src="https://github.com/user-attachments/assets/5a688567-ddb9-4440-9721-ecb7771d4ad8">

I created a ticket to investigate and fix the issue. I don’t think it is super high priority, but would be nice to have. We can use the workaround for now (and then clean it up in the new ticket). bcgov/entity#24754